Documentation/sources/how-to-messaging-delegate.md

```diff
@@ -0,0 +1,147 @@
+
+A reference to the `AEPMessage` object can be obtained by calling `fullscreenMessage.getParent()` . An example of how to access the `Message` in the `onShow` delegate method can be seen below:
+
+```java
+@Override
+public void onShow(FullscreenMessage fullscreenMessage) {
+  Message message = (Message) fullscreenMessage.getParent();
+  System.out.println("message was shown: " + message.id);
+}
+```
+
+### Controlling when a message should be shown to the end user
+
+If a custom  `FullscreenMessageDelegate` has been set in the `ServiceProvider`, this delegate's `shouldShowMessage` method will be called prior to displaying an in-app message for which the end user has qualified. You are responsible for returning `true` if the message should be shown, or `false` if the message should be suppressed.
+
+An example of when you may choose to suppress an in-app message due to the status of some other workflow within the app can be seen below:
+
+```java
+@Override
+public boolean shouldShowMessage(FullscreenMessage fullscreenMessage) {
+   if (someOtherWorkflowStatus == "inProgress") {
+        return false;
+    }
+    return true;
+}
+```
+
+Another option is to store a reference to the  `FullscreenMessage` object, and call the `show()` method on it at a later time.
+
+Continuing with the above example, after you have stored the message that was triggered initially, you can choose to show it upon completion of the other workflow:
+
+```java
+Message currentMessage = null;
+String anotherWorkflowStatus;
+
+public void otherWorkflowFinished() {
+    anotherWorkflowStatus = "complete";
+    currentMessage.show();
+}
+
+@Override
+public boolean shouldShowMessage(FullscreenMessage fullscreenMessage) {
+   if (someOtherWorkflowStatus.equals("inProgress")) {
+     // store the current message for later use
+     currentMessage = (Message) fullscreenMessage.getParent();
+     return false;
+   }
+  
+  return true;
+}
+```
+
+### Integrating the message into an existing UI
+
+If you would like to manually integrate the `View` that contains the UI for an in-app message, you can do so by accessing the `WebView` directly in a `MessagingDelegate` method.  
+
+In the example below, you can decide whether or not the in-app message should be directly integrated into your existing UI. If so, you capture a reference to the message's `WebView` and return `false` to prevent the message from being shown by the SDK:
+
+```java
+private Message currentMessage = null;
+private boolean shouldIntegrateMessageDirectly = true;
+private MessageWebView inAppMessageView;
+
+@Override
+public boolean shouldShowMessage(FullscreenMessage fullscreenMessage) {
+  if (shouldIntegrateMessageDirectly) {
+    this.currentMessage = (Message) fullscreenMessage.getParent();
+    
+    // cast to MessageWebView to access the startInAppMessage function
+    inAppMessageView = (MessageWebView) currentMessage.getWebView();
+    
+    return false;
+  }
+  
+  return true;
+}
+```
```

Comment:
i'm missing context from this example - unsure what `MessageWebView` is

Reply:
A `MessageWebView` object is an extension of an android `WebView`. I should add the call to startInAppMessage for more context.